src/local.rs

```diff
@@ -212,14 +212,106 @@ impl LocalAsset {
         Ok(())
     }
 
-    /// Copies an asset from one location on the local filesystem to another
+    /// Copies an asset from one location on the local filesystem to the given directory
+    ///
+    /// The destination will use the same file name as the origin has.
+    /// If you want to specify the destination file's name, use [`LocalAsset::copy_named`][].
     pub fn copy(
         origin_path: impl AsRef<Utf8Path>,
         dest_dir: impl AsRef<Utf8Path>,
     ) -> Result<PathBuf> {
         LocalAsset::load(origin_path)?.write(dest_dir)
     }
 
+    /// Copies an asset from one location on the local filesystem to another
+    ///
+    /// Both paths are assumed to be file names.
+    pub fn copy_named(
+        origin_path: impl AsRef<Utf8Path>,
+        dest_path: impl AsRef<Utf8Path>,
+    ) -> Result<()> {
+        let origin_path = origin_path.as_ref();
+        let dest_path = dest_path.as_ref();
+
+        fs::copy(origin_path, dest_path).map_err(|e| AxoassetError::LocalAssetCopyFailed {
+            origin_path: origin_path.to_string(),
+            dest_path: dest_path.to_string(),
+            details: Some(e),
+        })?;
+
+        Ok(())
+    }
+
+    /// Recursively copies a directory from one location to the given directory
+    ///
+    /// The destination will use the same dir name as the origin has, so
+    /// dest_dir is the *parent* of the copied directory. If you want to specify the destination's
+    /// dir name, use [`LocalAsset::copy_dir_named`][].
+    pub fn copy_dir(
+        origin_path: impl AsRef<Utf8Path>,
+        dest_dir: impl AsRef<Utf8Path>,
+    ) -> Result<Utf8PathBuf> {
+        let origin_path = origin_path.as_ref();
+        let dest_dir = dest_dir.as_ref();
+
+        let filename = Self::filename(origin_path)?;
+        let dest_path = dest_dir.join(filename);
+        Self::copy_dir_named(origin_path, &dest_path)?;
+        Ok(dest_path)
+    }
+
+    /// Recursively copies a directory from one location to another
+    ///
+    /// Both paths are assumed to be the names of the directory being copied
+    /// (i.e. dest_path is not the parent dir).
+    pub fn copy_dir_named(
+        origin_path: impl AsRef<Utf8Path>,
+        dest_path: impl AsRef<Utf8Path>,
+    ) -> Result<()> {
+        let origin_path = origin_path.as_ref();
+        let dest_path = dest_path.as_ref();
+
+        for entry in walkdir::WalkDir::new(origin_path) {
+            let entry = entry.map_err(|e| AxoassetError::LocalAssetCopyFailed {
+                origin_path: origin_path.to_string(),
+                dest_path: dest_path.to_string(),
+                details: e.into_io_error(),
+            })?;
+
+            let from = Utf8PathBuf::from_path_buf(entry.path().to_owned())
+                .map_err(|details| AxoassetError::Utf8Path { path: details })?;
+            let suffix = from.strip_prefix(origin_path).map_err(|_| {
```

Comment:
is there a reason to not fwd the error here?

Reply:
Error type mismatch (it's a camino error), but I think I'm gonna just flesh out error types here

Reply:
Oh actually as it turns out this is a std error that's just a marker type (empty struct, no actual context, so can be discarded).